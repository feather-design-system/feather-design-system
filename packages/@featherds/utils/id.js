const getSafeId = function (str) {
  str = str || "feather";
  const random = Math.floor(Math.random() * 1000000000);
  return [str.replace(/\s+/g, "-"), Date.now(), random].join("-");
};

export { getSafeId };
