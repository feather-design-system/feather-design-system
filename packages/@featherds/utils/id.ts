const getSafeId = function (str: string) {
  str = str || "feather";
  const random = Math.floor(Math.random() * 1000000000);
  return [str.replace(/\s+/g, "-"), Date.now(), random].join("-");
};

export { getSafeId };
