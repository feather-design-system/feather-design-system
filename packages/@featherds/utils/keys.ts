const KEYCODES = {
  PAGEUP: 33,
  PAGEDOWN: 34,
  UP: 38,
  DOWN: 40,
  LEFT: 37,
  RIGHT: 39,
  HOME: 36,
  END: 35,
  ENTER: 13,
  SPACE: 32,
  ESCAPE: 27,
  BACKSPACE: 8,
  DELETE: 46,
};

export { KEYCODES };
