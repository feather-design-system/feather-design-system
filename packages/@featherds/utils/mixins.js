var translateKey = {
  props: {
    translationKey: {
      type: String,
      default: "",
    },
  },
  methods: {
    translate: function (key) {
      return this.$t
        ? this.$t(this.translationKey + key)
        : "Vue i18n is not loaded";
    },
  },
};
export { translateKey };
