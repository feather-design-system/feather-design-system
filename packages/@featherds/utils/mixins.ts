const translateKey = {
  props: {
    translationKey: {
      type: String,
      default: "",
    },
  },
  methods: {
    translate: function (this: IComponent, key: string) {
      return this.$t
        ? this.$t(this.translationKey + key)
        : "Vue i18n is not loaded";
    },
  },
};

interface IComponent {
  translationKey: string;
  $t: (s: string) => void;
}
export { translateKey };
