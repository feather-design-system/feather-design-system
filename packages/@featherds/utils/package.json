{
  "name": "@featherds/utils",
  "version": "0.12.31",
  "publishConfig": {
    "access": "public"
  },
  "author": "NantHealth",
  "scripts": {
    "compile": "tsc --declaration"
  },
  "keywords": [
    "vue",
    "FeatherUtils"
  ]
}
