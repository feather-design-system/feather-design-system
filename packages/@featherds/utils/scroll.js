var toView = function (el, parent) {
  if (!el || !parent) {
    return;
  }
  var elHeight = el.getBoundingClientRect().height;
  const compStyle = getComputedStyle(el);
  elHeight += parseInt(compStyle["margin-top"], 10);
  elHeight += parseInt(compStyle["margin-bottom"], 10);
  parent.scrollTop =
    el.offsetTop - parent.getBoundingClientRect().height + elHeight;
};

export { toView };
