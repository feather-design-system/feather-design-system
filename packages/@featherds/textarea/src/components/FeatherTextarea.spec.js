import { nextTick } from "vue";
import axe from "@featherds/utils/test/axe";
import FeatherTextarea from "./FeatherTextarea.vue";
import { mount, config } from "@vue/test-utils";
import * as id from "@featherds/utils/id";
import "@featherds/input-helper/test/MutationObserver";

config.renderStubDefaultSlot = true;

jest.spyOn(id, "getSafeId").mockImplementation((x) => x);

const getWrapper = function (options) {
  return mount(FeatherTextarea, options);
};

describe("FeatherTextarea.vue", () => {
  it("should render an textarea with a label", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea with a label and hint text", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        hint: "hint text",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea with a preset value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        hint: "hint text",
        modelValue: "YEET",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea with raised label when focused", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    await wrapper.find("textarea").trigger("focus");
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea with raised label when focused and has a value", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
    });
    await wrapper.find("textarea").trigger("focus");
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should trigger textarea event when something is typed", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
      },
    });
    const textarea = wrapper.find("textarea");
    const expected = "TEEST";
    textarea.element.value = expected;
    textarea.trigger("input");
    expect(wrapper.emitted("update:modelValue")[1][0]).toBe(expected);
  });

  it("should render an textarea error", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea error and has focus", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
      },
    });
    await wrapper.find("textarea").trigger("focus");
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render an textarea error and has a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        error: "Error text",
        modelValue: "Test",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render a clear icon when clear has a value and textarea also has a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        clear: "clear",
        modelValue: "Test",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should not render a clear icon when the textarea doesn't have a value", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        clear: "clear",
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should render a maxlength attribute and character count when maxlength specified", () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "test",
        maxlength: 5,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("should clear textarea when clear-icon is clicked", async () => {
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
        clear: "clear",
      },
    });
    wrapper.vm.handleClear();
    await nextTick();
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.emitted("update:modelValue")[1][0]).toBe("");
  });

  it("should allow custom focus events", async () => {
    const fn = jest.fn();
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
      attrs: {
        onFocus: fn,
      },
    });

    await wrapper.find("textarea").trigger("focus");
    expect(fn).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  });

  fit("should allow custom blur events", () => {
    const fn = jest.fn();
    const wrapper = getWrapper({
      props: {
        label: "label",
        modelValue: "YEET",
      },
      attrs: {
        onBlur: fn,
      },
    });

    wrapper.find("textarea").trigger("blur");

    expect(fn).toHaveBeenCalled();
    expect(wrapper.element).toMatchSnapshot();
  });

  describe("a11y", () => {
    const accessibilityTest = async (options) => {
      const wrapper = mount(FeatherTextarea, options);
      expect(await axe(wrapper.element)).toHaveNoViolations();
    };
    it("should be accessible", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
        },
      });
    });
    it("should be accessible with hint", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
          hint: "HINTTT",
        },
      });
    });
    it("should be accessible with error", async () => {
      await accessibilityTest({
        props: {
          label: "label",
          modelValue: "YEET",
          error: "Error",
        },
      });
    });
  });
});
