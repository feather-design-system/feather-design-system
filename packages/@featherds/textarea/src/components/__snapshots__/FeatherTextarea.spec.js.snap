// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatherTextarea.vue should allow custom blur events 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should allow custom focus events 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should clear textarea when clear-icon is clicked 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should not render a clear icon when the textarea doesn't have a value 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render a clear icon when clear has a value and textarea also has a value 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <a
          class="action-icon hide-when-disabled"
          data-ref-id="feather-form-element-clear"
          data-v-051667c2=""
          data-v-d1dfbdf8=""
          href="#"
          title="clear"
        >
          <svg
            aria-hidden="true"
            class="feather-icon"
            data-v-0aa06a12=""
            data-v-d1dfbdf8=""
            focusable="false"
            role="img"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19,17.55,13.43,12,19,6.4A1,1,0,1,0,17.58,5L12,10.58,6.42,5A1,1,0,0,0,5,6.42L10.6,12,5,17.6A1,1,0,0,0,5,19a1,1,0,0,0,.71.29,1,1,0,0,0,.71-.3L12,13.41,17.6,19a1,1,0,0,0,.71.29A1,1,0,0,0,19,19,1,1,0,0,0,19,17.55Z"
            />
          </svg>
        </a>
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render a maxlength attribute and character count when maxlength specified 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="5"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    <div
      class="feather-input-count"
      data-ref-id="feather-form-element-count"
    >
      4 / 5
    </div>
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea error 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container error feather-input-content error"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-textarea error"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-v-051667c2=""
          data-v-0aa06a12=""
          data-v-41337cd6=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-cef1410a=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea error and has a value 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised error feather-input-content error"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-textarea error"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-v-051667c2=""
          data-v-0aa06a12=""
          data-v-41337cd6=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-cef1410a=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea error and has focus 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised focused error feather-input-content error focused"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="true"
        class="feather-textarea error"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <svg
          aria-hidden="true"
          class="error-icon hide-when-disabled feather-icon error-icon hide-when-disabled"
          data-v-051667c2=""
          data-v-0aa06a12=""
          data-v-41337cd6=""
          focusable="false"
          role="img"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12,2A10,10,0,1,0,22,12,10,10,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8,8,0,0,1,12,20Z"
          />
          <rect
            height="7"
            rx="1"
            width="2"
            x="11"
            y="7"
          />
          <rect
            height="2"
            rx="0.65"
            width="2"
            x="11"
            y="15"
          />
        </svg>
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <div
      aria-live="assertive"
      class="feather-input-error"
      data-ref-id="feather-form-element-error"
      data-v-cef1410a=""
    >
      Error text
    </div>
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea with a label 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea with a label and hint text 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 16px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <div
      class="feather-input-hint"
      data-ref-id="feather-form-element-hint"
      data-v-cef1410a=""
    >
      hint text
    </div>
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea with a preset value 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised feather-input-content"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <div
      class="feather-input-hint"
      data-ref-id="feather-form-element-hint"
      data-v-cef1410a=""
    >
      hint text
    </div>
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea with raised label when focused 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;

exports[`FeatherTextarea.vue should render an textarea with raised label when focused and has a value 1`] = `
<div
  class="feather-textarea-container"
>
  <div
    class="feather-input-wrapper-container raised focused feather-input-content focused"
    data-v-051667c2=""
  >
    <fieldset
      aria-hidden="true"
      class="feather-input-border"
      data-v-051667c2=""
    >
      <legend
        data-v-051667c2=""
      >
        label
      </legend>
    </fieldset>
    <label
      class="feather-input-label"
      data-ref-id="feather-form-element-label"
      data-v-051667c2=""
      for="feather-textarea-label"
      style="left: 12px;"
    >
      label
    </label>
    <div
      class="feather-input-wrapper"
      data-v-051667c2=""
    >
      <div
        class="prefix"
        data-v-051667c2=""
      >
        
        
      </div>
      
      <textarea
        aria-describedby="feather-textarea-description"
        aria-disabled="false"
        aria-invalid="false"
        class="feather-textarea"
        data-ref-id="feather-input"
        id="feather-textarea-label"
        maxlength="false"
        name="feather-textarea-label"
      />
      
      <div
        class="post"
        data-v-051667c2=""
      >
        <!---->
        <!---->
        
        
      </div>
    </div>
  </div>
  <div
    class="feather-input-sub-text"
    data-v-cef1410a=""
    id="feather-textarea-description"
  >
    <!---->
    <!---->
    
    
  </div>
</div>
`;
